private/buf/buffetch/source_ref.go

```diff
@@ -24,22 +24,55 @@ import (
 var _ SourceRef = &sourceRef{}
 
 type sourceRef struct {
-	bucketRef internal.BucketRef
-	dirPath   string
+	bucketRef  internal.BucketRef
+	dirPath    string
+	subDirPath string
 }
 
 func newSourceRef(bucketRef internal.BucketRef) *sourceRef {
 	var dirPath string
 	if dirRef, ok := bucketRef.(internal.DirRef); ok {
 		dirPath = dirRef.Path()
 	}
+	var subDirPath string
+	if archiveRef, ok := bucketRef.(internal.ArchiveRef); ok {
+		subDirPath = archiveRef.SubDirPath()
+	}
+	if subDirPath == "" {
```

Comment:
if anything I'd want to error if both are true, not silently say that `ArchiveRef` takes precedence

Reply:
Discussed offline - we should just add a new type and not use `newSourceRef` for the `ArchiveRef` and the `GitRef` types.